pkg/topology/topology.go

```diff
@@ -28,3 +28,9 @@ type PeerAdder interface {
 type ClosestPeerer interface {
 	ClosestPeer(addr swarm.Address) (peerAddr swarm.Address, err error)
 }
+
+// ScoreFunc is implemented by components that need to score peers in a different way than XOR distance.
+type ScoreFunc func(peer swarm.Address) (score float32)
+
+// EachPeerFunc is a callback that is called with a peer and its PO
+type EachPeerFunc func(swarm.Address, uint8) (stop, jumpToNext bool, err error)
```

Comment:
EachPeerFunc is more related to kademlia as it has the concept of proximity order, while topology package is currently unaware of it. I think that it is ok to have it in kademlia or pslice package. WDYT?

Reply:
yeah, i put it here because if we still want to keep the full connectivity thing in the codebase it would have to have the same interface as kademlia with the iterators, then this function will belong here if the type reference is shared.

Reply:
That is ok, thanks.